app/scripts/modules/kubernetes/serverGroup/configure/CommandBuilder.js

```diff
@@ -54,11 +48,50 @@ module.exports = angular.module('spinnaker.kubernetes.serverGroupCommandBuilder.
         command.interestingHealthProviderNames = ['Kubernetes'];
       }
 
-      attemptToSetValidCredentials(application, defaultCredentials, command);
+      attemptToSetValidAccount(application, defaultAccount, command);
 
       return $q.when(command);
     }
 
+    function findUpstreamImages(current, all, visited = {}) {
+      // This actually indicates a loop in the stage dependencies.
+      if (visited[current.refId]) {
+        return [];
+      } else {
+        visited[current.refId] = true;
+      }
+      let result = [];
+      if (current.type === 'findImage') {
+        result.push(current.name);
+      }
+      current.requisiteStageRefIds.forEach(function(id) {
+        let next = _.filter(all, function(stage) {
+          return stage.refId === id;
+        });
+        if (next) {
```

Comment:
I think `next` will always evaluate to a truthy value, even if nothing is found (it'll be an empty array).

If you rewrite the filter above as:
```javascript
let [next] = all.filter((stage) => stage.refId === id);
```
it'll set `next` to the first item in the result, or `null` if the result of the filter is an empty array, and this loop would work as expected.

I could be wrong on this, but test it out either way to make sure.